### ISSUE TITLE
When syncing a source to a destination not all columns get copied

### ISSUE THREAD

**Issue by antonioorct** [jitsucom/jitsu#1040]
**Version:** Jitsu Next @ latest

Source postgres table:  
![image](https://github.com/jitsucom/jitsu/assets/14859262/6a6e14c4-8cd0-4f11-a3f2-5620407cb891)

Destination postgres table:  
![image](https://github.com/jitsucom/jitsu/assets/14859262/d155656c-613b-41ce-9fe7-c694837afc47)

The source postgres table contains only one row whose `scope` and `workspaceId` fields are null (these fields are missing from the destination table). So it seems to be only happening when a column in all rows are `null`.

I assume this is an issue with Airbyte but I still want to check in.

**Comment by antonioorct** [comment]
Going into the `json_schema` fetched after a call to `/discover` I get the following for this source table:  
```json
{...
    "json_schema": {
        "type": "object",
        "properties": {
            "id": {
                "type": "string"
            },
            "scope": {
                "type": "string"
            },
            "userId": {
                "type": "string"
            },
            "createdAt": {
                "type": "string",
                "format": "date-time",
                "airbyte_type": "timestamp_without_timezone"
            },
            "updatedAt": {
                "type": "string",
                "format": "date-time",
                "airbyte_type": "timestamp_without_timezone"
            },
            "preferences": {
                "type": "string"
            },
            "workspaceId": {
                "type": "string"
            }
        }
    }
}
```

I'm not sure if it's related that none of the fields have got any property to indicate `nullable`.

**Comment by absorbb** [comment]
Look like something that we can fix on our side. We will take a look

**Comment by vklimontovich** [comment]
@absorbb is this still an issue?

**Comment by vklimontovich** [comment]
It's not a bug, it's how Jitsu is designed at the moment. The connector sends data to [Bulker](https://github.com/jitsucom/bulker) as an array (stream) of JSON object, and Bulker omits all `null` values. Also, it create columns on demand, so as a result the columns are not being created. But once the column will have a non-null value at least in one record coming in, it will be created.

That being said, I agree that the current design is not ideal. We will make a change that will remove `null` value deletion if the data is coming from Jitsu Connectors

**Comment by absorbb** [comment]
~~Fixed in `jitsucom/bulker:v1.0.1`~~

**Comment by absorbb** [comment]
Fixed in Jitsu 2.6.0
